### PR TITLE
Refactor helm status command tests to remove duplication

### LINKED ISSUE #3659: Possible duplication of test code in helm status command tests
The tests for the helm status command seem to duplicate functionality very similar to `runReleaseCases` function:

https://github.com/kubernetes/helm/blob/2c51155724940eec3bc8cb97e38ffbf4db6a0172/cmd/helm/status_test.go#L134-L152

### PR DESCRIPTION
**Pull request by adshmh**
fix(helm) refactor helm status command tests to use releaseCase struct and the corresponding function runReleaseCases. Fixes #3659.

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

### CODE REVIEW

**Review by thomastaylor312 — APPROVED** [review]
LGTM. Thank you for all of the refactors! Waiting on a second review because it is marked as a large

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
nice!

**Pull request merged**
